File: PoissonJuliaLanguage/PoissonDisk.jl
#Simulate a Poisson point process on a disk
#Author: H. Paul Keeler, 2019.

#Note: Need the .+ for adding a scalar to an array
#Also need . for sqrt, exp, cos, sin etc and assinging scalars to arrays

using Distributions #for random simulations
using Plots #for plotting

#Simulation window parameters
r=1; #radius of disk
xx0=0; yy0=0; #centre of disk
areaTotal=pi*r^2; #area of disk

#Point process parameters
lambda=100; #intensity (ie mean density) of the Poisson process

#Simulate Poisson point process
numbPoints=rand(Poisson(areaTotal*lambda)); #Poisson number of points
theta=2*pi*(rand(numbPoints,1));#angular  of Poisson points
rho=r*sqrt.(rand(numbPoints,1));#radial coordinates of Poisson points

#Convert polar to Cartesian coordinates
xx=rho.*cos.(theta);
yy=rho.*sin.(theta);

#Shift centre of disk to (xx0,yy0)
xx=xx.+xx0;
yy=yy.+yy0;

plot1=scatter(xx,yy,xlabel ="x",ylabel ="y", leg=false);
display(plot1);

File: PoissonDisk/PoissonDisk.R
#Simulate a Poisson point process on a disk
#Author: H. Paul Keeler, 2018.

#Simulation window parameters
r=1; #radius of disk
xx0=0; yy0=0; #centre of disk
areaTotal=pi*r^2; #area of disk

#Point process parameters
lambda=100; #intensity (ie mean density) of the Poisson process

#Simulate Poisson point process
numbPoints=rpois(1,areaTotal*lambda);#Poisson number of points
theta=2*pi*runif(numbPoints);#angular  of Poisson points
rho=r*sqrt(runif(numbPoints));#radial coordinates of Poisson points

#Convert from polar to Cartesian coordinates
xx=rho*cos(theta);
yy=rho*sin(theta);

#Shift centre of disk to (xx0,yy0)
xx=xx+xx0;
yy=yy+yy0;

#Plotting
par(pty="s"); #use square plotting region
plot(xx,yy,'p',xlab='x',ylab='y',col='blue');

#Simulate a Poisson point process with spatstat library
library("spatstat");
ppPoisson<-rpoispp(lambda,win=disc(radius=r,centre=c(xx0,yy0))) #create Poisson "point pattern" object
plot(ppPoisson); #Plot point pattern object
#retrieve x/y values from point pattern object
xx=ppPoisson$x; 
yy=ppPoisson$y;

#do the above, but in one line
plot(ppPoisson<-rpoispp(lambda,win=disc(radius=r,centre=c(xx0,yy0))) ); #create and plot point pattern object
